File: UnderTheKitchen-OC/UnderTheKitchen-OC/Classes/Common/TabBar/UTKCommonController.m
//
//  UTKCommonController.m
//  UnderTheKitchen-OC
//
//  Created by jyapp on 2018/8/17.
//  Copyright © 2018年 wsl. All rights reserved.
//

#import "UTKCommonController.h"

@interface UTKCommonController ()

@end

@implementation UTKCommonController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    // Do any additional setup after loading the view.
}






- (void)setRightTitleButtonTitle:(NSString *)title color:(UIColor *)color image:(UIImage *)image action:(SEL)action{
    UIButton *bt  = [UIButton buttonWithType:UIButtonTypeCustom];
    [bt setTitle:title forState:UIControlStateNormal];
    [bt setImage:image forState:UIControlStateNormal];
    [bt setTitle:title forState:UIControlStateHighlighted];
    [bt setImage:image forState:UIControlStateHighlighted];
    [bt setTitleColor:color forState:UIControlStateNormal];
    bt.titleLabel.font = [UIFont systemFontOfSize:15];
    
    [bt addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
    [bt sizeToFit];
    bt.imageEdgeInsets = UIEdgeInsetsMake(5, 0, -5, 0);
    UIBarButtonItem *btitem = [[UIBarButtonItem alloc] initWithCustomView:bt];
    self.navigationItem.rightBarButtonItems = @[btitem];
}

- (void)setRightTitleButtonsImage:(UIImage *)image action:(SEL)action withImage:(UIImage *)withimage withaction:(SEL)withaction {
    
    UIView *view =[[UIView alloc] initWithFrame:CGRectMake(0, 0, 80, 44)];
    UIButton *bt  = [UIButton buttonWithType:UIButtonTypeCustom];
    [bt setImage:image forState:UIControlStateNormal];
    [bt setImage:image forState:UIControlStateHighlighted];
    bt.titleLabel.font = [UIFont systemFontOfSize:15];
    [bt setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [bt addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
    [bt sizeToFit];
    bt.frame = CGRectMake(0, 0, 40, 44);
    //UIBarButtonItem *btitem = [[UIBarButtonItem alloc] initWithCustomView:bt];
//
//    UIBarButtonItem *negativeSpacer = [[UIBarButtonItem alloc]
//                                       initWithBarButtonSystemItem:UIBarButtonSystemItemFixedSpace
//                                       target:nil action:nil];
//    negativeSpacer.width = 15;
    
    UIButton *withbt  = [UIButton buttonWithType:UIButtonTypeCustom];
    [withbt setImage:withimage forState:UIControlStateNormal];
    [withbt setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [withbt setImage:withimage forState:UIControlStateHighlighted];
    withbt.titleLabel.font = [UIFont systemFontOfSize:15];
    [withbt addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
    [withbt sizeToFit];
    withbt.frame = CGRectMake(80-40, 0, 40, 44);
    
    [view addSubview:bt];
    [view addSubview:withbt];
    
    UIBarButtonItem *withbtitem = [[UIBarButtonItem alloc] initWithCustomView:view];
    self.navigationItem.rightBarButtonItems =@[withbtitem];
}


- (void)setLeftTitleButtonTitle:(NSString *)title color:(UIColor *)color image:(UIImage *)image position:(UTKImagePosition)position action:(SEL)action{
    UIButton *bt  = [UIButton buttonWithType:UIButtonTypeCustom];
    [bt setTitle:title forState:UIControlStateNormal];
    [bt setImage:image forState:UIControlStateNormal];
    [bt setTitle:title forState:UIControlStateHighlighted];
    [bt setImage:image forState:UIControlStateHighlighted];
    [bt setTitleColor:color forState:UIControlStateNormal];
    bt.titleLabel.font = [UIFont systemFontOfSize:15];
    [bt addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
    [bt setImagePosition:position spacing:1.0];
    [bt sizeToFit]; bt.imageEdgeInsets = UIEdgeInsetsMake(5, 0, -5, 0);
    UIBarButtonItem *btitem = [[UIBarButtonItem alloc] initWithCustomView:bt];
    self.navigationItem.leftBarButtonItems = @[btitem];
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: UnderTheKitchen-Swift/UnderTheKitchen-Swift/Classes/Common/TabBar/UTKCommonController.swift
//
//  UTKCommonController.swift
//  UnderTheKitchen-Swift
//
//  Created by jyapp on 2018/8/23.
//  Copyright © 2018年 wsl. All rights reserved.
//

import UIKit

class UTKCommonController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.white

        // Do any additional setup after loading the view.
    }

    
    func setRightTitleButtonTitle(title:String,color:UIColor,image:UIImage,action:Selector ){
        let bt = UIButton()
        bt.setTitle(title, for: UIControlState.normal)
        bt.setImage(image, for: UIControlState.normal)
        bt.setTitle(title, for: UIControlState.highlighted)
        bt.setImage(image, for: UIControlState.highlighted)
        bt .setTitleColor(color, for: UIControlState.normal)
        bt.titleLabel?.font = UIFont.systemFont(ofSize: 15)
        bt.sizeToFit()
        bt.addTarget(self, action:action , for: UIControlEvents.touchUpInside)
        let btitem = UIBarButtonItem.init(customView: bt)
        self.navigationItem.rightBarButtonItems = [btitem]
    }
    
    func setRightTitleButtonsImage(image :UIImage,action :Selector,withimage:UIImage,withaction:Selector ){
        
        let customView = UIView.init(frame: CGRect.init(x: 0, y: 0, width: 80, height: 44))
        let bt = UIButton()
        bt.setImage(image, for: UIControlState.normal)
        bt.setImage(image, for: UIControlState.highlighted)
        bt .setTitleColor(UIColor.black, for: UIControlState.normal)
        bt.titleLabel?.font = UIFont.systemFont(ofSize: 15)
        bt.sizeToFit()
        bt.addTarget(self, action:action , for: UIControlEvents.touchUpInside)
        bt.frame=CGRect.init(x: 0, y: 0, width: 40, height: 44)
        
        let withbt = UIButton()
        withbt.setImage(withimage, for: UIControlState.normal)
        withbt.setImage(withimage, for: UIControlState.highlighted)
        withbt .setTitleColor(UIColor.black, for: UIControlState.normal)
        withbt.titleLabel?.font = UIFont.systemFont(ofSize: 15)
        withbt.sizeToFit()
        withbt.addTarget(self, action:withaction , for: UIControlEvents.touchUpInside)
        withbt.frame=CGRect.init(x: 80-40, y: 0, width: 40, height: 44)
        
        customView.addSubview(bt)
        customView.addSubview(withbt)
        
        let withbtitem = UIBarButtonItem.init(customView: customView)
        self.navigationItem.rightBarButtonItems = [withbtitem]
    }
    
    
    
    
    func setLeftTitleButtonTitle(title:String,color:UIColor,image:UIImage,position:UTKImagePosition, action:Selector ){
        let bt = UIButton()
        bt.setTitle(title, for: UIControlState.normal)
        bt.setImage(image, for: UIControlState.normal)
        bt.setTitle(title, for: UIControlState.highlighted)
        bt.setImage(image, for: UIControlState.highlighted)
        bt .setTitleColor(color, for: UIControlState.normal)
        bt.titleLabel?.font = UIFont.systemFont(ofSize: 15)
        bt.setImagePosition(position: position , spacing: 1.0)
        bt.sizeToFit()
        bt.addTarget(self, action:action , for: UIControlEvents.touchUpInside)
        let btitem = UIBarButtonItem.init(customView: bt)
        self.navigationItem.leftBarButtonItems = [btitem]
    }
    
    
    
    
    
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
